mailbox/api/src/test/java/org/apache/james/mailbox/events/EventDeadLettersHealthCheckContract.java

```diff
@@ -0,0 +1,130 @@
+
+    EventDeadLetters eventDeadLetters();
+
+    @Test
+    default void checkShouldReturnHealthyWhenEventDeadLetterEmpty() {
+        assertThat(testee().check().isHealthy()).isTrue();
+        assertThat(testee().check())
+            .isEqualTo(Result.healthy(COMPONENT_NAME));
+    }
+
+    @Test
+    default void checkShouldReturnDegradedWhenEventDeadLetterContainEvent() {
+        eventDeadLetters().store(GROUP_A, EVENT_1, INSERTION_ID_1).block();
+
+        assertThat(testee().check().isDegraded()).isTrue();
+        assertThat(testee().check())
+            .isEqualTo(Result.degraded(COMPONENT_NAME, "EventDeadLetters contain events"));
+    }
+
+    @Test
+    default void checkShouldReturnDegradedWhenEventDeadLetterContainEvents() {
+        eventDeadLetters().store(GROUP_A, EVENT_1, INSERTION_ID_1).block();
+        eventDeadLetters().store(GROUP_B, EVENT_2, INSERTION_ID_2).block();
+
+        assertThat(testee().check().isDegraded()).isTrue();
+        assertThat(testee().check())
+            .isEqualTo(Result.degraded(COMPONENT_NAME, "EventDeadLetters contain events"));
+    }
+
+    @Test
+    default void checkShouldReturnHealthyWhenRemovedAllEventDeadLetters() {
+        eventDeadLetters().store(GROUP_A, EVENT_1, INSERTION_ID_1).block();
+        eventDeadLetters().store(GROUP_B, EVENT_2, INSERTION_ID_2).block();
+
+        assertThat(testee().check().isDegraded()).isTrue();
+        assertThat(testee().check())
+            .isEqualTo(Result.degraded(COMPONENT_NAME, "EventDeadLetters contain events"));
+
+        eventDeadLetters().remove(GROUP_A, INSERTION_ID_1).block();
+        eventDeadLetters().remove(GROUP_B, INSERTION_ID_2).block();
+
+        assertThat(testee().check().isHealthy()).isTrue();
+        assertThat(testee().check())
+            .isEqualTo(Result.healthy(COMPONENT_NAME));
+    }
+
+    @Test
+    default void checkShouldReturnDegradedWhenRemovedSomeEventDeadLetters() {
+        eventDeadLetters().store(GROUP_A, EVENT_1, INSERTION_ID_1).block();
+        eventDeadLetters().store(GROUP_B, EVENT_2, INSERTION_ID_2).block();
+
+        assertThat(testee().check().isDegraded()).isTrue();
+        assertThat(testee().check())
+            .isEqualTo(Result.degraded(COMPONENT_NAME, "EventDeadLetters contain events"));
+
+        eventDeadLetters().remove(GROUP_A, INSERTION_ID_1).block();
+
+        assertThat(testee().check().isDegraded()).isTrue();
+        assertThat(testee().check())
+            .isEqualTo(Result.degraded(COMPONENT_NAME, "EventDeadLetters contain events"));
+    }
+
+    @Test
+    default void checkShouldReturnUnHealthyWhenEventDeadLetterError() {
+        Result actualResult;
+        try {
+            createErrorWhenDoingHealthCheck();
+            actualResult = testee().check();
+        } finally {
+            resolveErrorWhenDoingHealthCheck();
+        }
+
+        assertThat(actualResult.isUnHealthy()).isTrue();
+    }
+
+    void createErrorWhenDoingHealthCheck();
```

Comment:
Please put these methods to implement at the to of the contract (non default methods goes first)